students/scott_bromley/lesson09/mailroom_oo/donor_models.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env python3
+
+from datetime import datetime
+from statistics import mean
+from re import match
+
+
+class Donor:
+    """
+    Donor class represents a single donor
+    """
+    def __init__(self, donor_name="", donations=None):
+        """
+        Donor object initialization
+        :param donor_name: full name of donor
+        :param donations: donations
+        """
+        self._donor_name = donor_name
+        self.donor_name = donor_name if donor_name is not None else ""
+        self._donations = []
+        if not donations:
+            self._donations = []
+        elif hasattr(donations, "__iter__"):
+            try:
+                [self.add_donation(donation) for donation in donations]
```

Comment:
this is not a valid use case for a comprehension, the result should be stored in a variable.

this should be just a regular for loop